js/navigation.js

```diff
@@ -111,4 +125,104 @@ searchResultsListDiv.addEventListener('click', function (event) {
   }
 });
 
-function makeShoppintList() {}
+function makeShoppingList() {
+  var list = shoppingList.listIngredients();
+  var listId = document.getElementById('shopping-list');
+  // remove all children from container
+  while (listId.firstChild) {
+    listId.removeChild(listId.firstChild);
+  }
+  for (var i = 0; i < list.length; i++) {
+    // dom create heading for label and dom create ul
+    var unorderdList = document.createElement('ul');
+
+    var recipeName = document.createElement('h2');
+    var recipeNameText = document.createTextNode(list[i].label);
+    recipeName.appendChild(recipeNameText);
+
+    recipeName.appendChild(unorderdList);
+
+    for (var il = 0; il < list[i].ingredients.length; il++) {
+      // dom create li's and append to ul
+      var liList = document.createElement('li');
+      var liText = document.createTextNode(list[i].ingredients[il]);
+      liList.appendChild(liText);
+      unorderdList.appendChild(liList);
+    }
+    // append heading and ul to container
+    listId.append(recipeName, unorderdList);
+  }
+}
+
+function makeFavoritesList() {
+  var list = favorites.listRecipes();
+
+  while (favResultsDiv.firstChild) {
+    favResultsDiv.removeChild(favResultsDiv.firstChild);
+  }
+
+  for (let i = 0; i < list.length; i++) {
+    var p = document.createElement('p');
+    p.onclick = function () {
+      window.open(list[i].recipe.url);
+    };
+    var pText = document.createTextNode(list[i].recipe.label);
+    p.appendChild(pText);
+
+    var favStar = document.createElement('i');
+
+    if (favorites.getRecipe(list[i].recipe.uri) !== undefined) {
+      favStar.className = 'fas fa-star favorite';
+    } else {
+      favStar.className = 'far fa-star favorite';
+    }
+    favStar.setAttribute('data-recipe-uri', list[i].recipe.uri);
+
+    var cartIcon = document.createElement('i');
+
+    if (shoppingList.getRecipe(list[i].recipe.uri) !== undefined) {
+      cartIcon.className = 'fas fa-shopping-cart shopping-cart';
+    } else {
+      cartIcon.className = 'fas fa-cart-plus shopping-cart';
+    }
+    cartIcon.setAttribute('data-recipe-uri', list[i].recipe.uri);
+
+    var divBackgroundImg = document.createElement('div');
+    divBackgroundImg.setAttribute('class', 'img-background');
+
+    divBackgroundImg.style.backgroundImage =
+      'url(' + list[i].recipe.image + ')';
+
+    var divRecipe = document.createElement('div');
+    divRecipe.setAttribute('class', 'recipe');
+
+    divRecipe.append(favStar, cartIcon, divBackgroundImg, p);
+
+    favResultsDiv.append(divRecipe);
+  }
+}
```

Comment:
You shouldn't be adding elements to the DOM in the same function they are generated, you should have a function that takes in an object with the pertinent data, and returns the DOM tree for that object's data. In this function, you could call that function for each entry in your list. I recommend reviewing `dom-creation` for a reminder of the format I am referencing.